### ISSUE TITLE
changes to SDL

### ISSUE THREAD

**Issue by sorenbs** [graphcool/framework#1380]
In addition to the changes specified in [make Embedded Types Obsolete](https://github.com/graphcool/framework/issues/1160#issuecomment-345373277) we want to introduce the following changes:

- rename `@isUnique` to `@unique`
- rename `@defaultValue` to `@default`
- remove `@model`

**Comment by kbrandwijk** [comment]
I'm going to revive an older dicussion from an issue that is now closed: I think `@default` is not intuitive, as I don't except to provide a value, based on the naming. That's the difference between `@unique` and `@defaultValue`. The first doesn't expect a value, the second does.

**Comment by sorenbs** [comment]
So you actually prefer `@defaultValue`?

@schickling was the main driver on this change, so I will defer to him :-)

The main argument is that writing value twice feels redundant: 

`@defaultValue(value: "my default value")`

**Comment by schickling** [comment]
I still think that `@default` is better. We should solve this by improving the (IDE) tooling to give you feedback that `@default` is missing the `value` argument.

**Comment by mkozhukharenko** [comment]
`default` is better, I think

- - -

what is the sense in removing `migrationValue` ?

- - -

Are these schemas different?

```
type Article {
  isPublished: Boolean @default(value: "false")
}
type Article {
  isPublished: Boolean  # @default(value: false)
}
```
latests examples has `#` character before a directive

**Comment by marktani** [comment]
* `@migrationValue` will be removed once #1263 is in place.
* `#` marks the rest of the row as comment. So yes, these schemas are different.

**Comment by LFP6** [comment]
I noticed #1489 / 315d880 was merged with work on optional @relations and optional backreferences, and that todo item is checked off here. However, while the prior is in the latest beta the latter is not - is this still in progress? Or was the schema validation just not removed?

**Comment by marktani** [comment]
Do you have a specific example that helps understand what exactly you mean?

Here's an example of both optional `@relation` directives as well as unidirectional relations. You can already deploy the following datamodel in the latest beta:

```graphql
type User {
  id: ID! @unique
  name: String!
  location: Location
  posts: [Post!]!
}

type Location {
  lng: Float!
  lat: Float!
}

type Post {
  id: ID! @unique
  title: String!
  author: User!
}
```

This enables you a query like this:

```graphql
query {
  users {
    id
    location {
      lng
      lat
    }
    posts {
      id
      author {
        id
      }
    }
  }
}
```

Note:

* unidiretional relations: on `location`, there is no `user` field (because `Location` doesn't define the back relation).
* optional `@relation` directive: `Post.user` and `User.posts` works, but we did not specify a `@relation` directive, because there is only one relation between those two types. If there were at least 2 relations, we'd need to use `@relation` to resolve the ambiguity.

**Comment by LFP6** [comment]
Sorry, I should have been more explicit. I was specifically speaking to one-sided named relationships failing with "A relation directive with a name must appear exactly 2 times", which does not correlate with what I see speced in https://github.com/graphcool/graphcool/issues/1160#issuecomment-345373277 and older issues.

**Comment by marktani** [comment]
This does sound unintended, could you create a separate issue with a detailed bug report? 🙂 Thanks!
